services/intersection-api/api/src/main/java/us/dot/its/jpo/ode/api/services/SNMPService.java

```diff
@@ -0,0 +1,169 @@
+package us.dot.its.jpo.ode.api.services;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.snmp4j.*;
+import org.snmp4j.event.ResponseEvent;
+import org.snmp4j.mp.MPv3;
+import org.snmp4j.mp.SnmpConstants;
+import org.snmp4j.security.AuthSHA;
+import org.snmp4j.security.PrivAES128;
+import org.snmp4j.security.SecurityLevel;
+import org.snmp4j.security.SecurityModels;
+import org.snmp4j.security.SecurityProtocols;
+import org.snmp4j.security.USM;
+import org.snmp4j.security.UsmUser;
+import org.snmp4j.smi.*;
+import org.snmp4j.transport.DefaultUdpTransportMapping;
+import org.springframework.stereotype.Service;
+
+import java.io.IOException;
+
+import javax.annotation.PostConstruct;
+
+@Service
+public class SNMPService {
+
+    private static final Logger logger = LoggerFactory.getLogger(SNMPService.class);
+
+    @PostConstruct
+    public void init() throws Exception {
+
+    }
+
+    // Some RSUs do not require authentication to retrieve information. They may
+    // respond to SNMP V2 requests.
+    public String getSnmpV2Value(String ipAddress, String community, String oid) throws Exception {
+
+        TransportMapping<UdpAddress> transport = new DefaultUdpTransportMapping();
+        Snmp snmp = new Snmp(transport);
+        transport.listen();
+
+        CommunityTarget<UdpAddress> target = new CommunityTarget<>();
+        target.setCommunity(new OctetString(community));
+        target.setAddress(new UdpAddress(ipAddress + "/161"));
+        target.setRetries(2);
+        target.setTimeout(1500);
```

Comment:
(blocking) these configurations should be set and configured in the `application.yaml` in case we need to adjust them